src/evict/evict_lru.c

```diff
@@ -147,36 +148,36 @@ __evict_list_clear(WT_SESSION_IMPL *session, WT_EVICT_ENTRY *e)
 {
     if (e->ref != NULL) {
         WT_ASSERT(session, F_ISSET_ATOMIC_16(e->ref->page, WT_PAGE_EVICT_LRU));
-        F_CLR_ATOMIC_16(e->ref->page, WT_PAGE_EVICT_LRU);
+        F_CLR_ATOMIC_16(e->ref->page, WT_PAGE_EVICT_LRU | WT_PAGE_IN_URGENT_QUEUE);
     }
     e->ref = NULL;
     e->btree = WT_DEBUG_POINT;
 }
 
 /*
- * __wt_evict_list_clear_page --
- *     Make sure a page is not in the LRU eviction list. This called from the page eviction code to
- *     make sure there is no attempt to evict a child page multiple times.
+ * __evict_list_clear_page_locked --
+ *     This function searches for the page on all the eviction queues (skipping the urgent queue if
+ *     requested) and clears it if found. It does not take the eviction queue lock, so the caller
+ *     should hold the appropriate locks before calling this function.
  */
-void
-__wt_evict_list_clear_page(WT_SESSION_IMPL *session, WT_REF *ref)
+static void
+__evict_list_clear_page_locked(WT_SESSION_IMPL *session, WT_REF *ref, bool exclude_urgent)
 {
     WT_CACHE *cache;
     WT_EVICT_ENTRY *evict;
-    uint32_t i, elem, q;
+    uint32_t elem, i, q, last_queue_idx;
     bool found;
 
-    WT_ASSERT(session, __wt_ref_is_root(ref) || ref->state == WT_REF_LOCKED);
-
-    /* Fast path: if the page isn't on the queue, don't bother searching. */
-    if (!F_ISSET_ATOMIC_16(ref->page, WT_PAGE_EVICT_LRU))
-        return;
-
+    last_queue_idx = exclude_urgent ? WT_EVICT_URGENT_QUEUE : WT_EVICT_QUEUE_MAX;
     cache = S2C(session)->cache;
-    __wt_spin_lock(session, &cache->evict_queue_lock);
-
     found = false;
-    for (q = 0; q < WT_EVICT_QUEUE_MAX && !found; q++) {
+
+    for (q = 0; q < last_queue_idx && !found; q++) {
+        if (exclude_urgent)
+            /* Ensure that the page in the urgent queue is not accidentally cleared. */
+            WT_ASSERT_ALWAYS(session, q != WT_EVICT_URGENT_QUEUE,
```

Comment:
@agorrod after @sulabhM comment [here](https://github.com/wiredtiger/wiredtiger/pull/8950#discussion_r1144194722), this check seems unnecessary. Don't you think?

Reply:
I agree

Reply:
Updated in [29b7443](https://github.com/wiredtiger/wiredtiger/pull/8950/commits/29b7443a51d342f5b9be7e0385400390374b8897)